### ISSUE TITLE
Fix OnyxSelect: Optimize screen reader access of internal checkboxes

### ISSUE THREAD

**Issue by BoppLi** [SchwarzIT/onyx#1026]
Non-critical Bug. Follow up to 
- #732

Steps to reproduce: Add the test code
```ts
  // ACT
  const accessibilityScanResults = await makeAxeBuilder().analyze();

  // ASSERT
  expect(accessibilityScanResults.violations).toEqual([]);
```
to the test case `'should interact with multiselect'` in `packages/sit-onyx/src/components/OnyxListbox/OnyxListbox.ct.tsx`

![Image](https://github.com/SchwarzIT/onyx/assets/151019977/e395b920-8ff8-4f6f-9183-de32a56410a9)
https://dequeuniversity.com/rules/axe/4.9/nested-interactive?application=playwright

One approach to fix the error is to always mark the inputs as disabled, which in return would complicate the usage of `packages/sit-onyx/src/styles/mixins/checkbox.scss` to still show the correct styles when enabled/disabled and checked/unchecked inside ListboxOptions.

![Image](https://github.com/SchwarzIT/onyx/assets/151019977/e3e85a26-f27f-4f0b-8739-1d36b1b81702)

**Comment by MajaZarkova** [comment]
Even if we use the disabled property as default, and adjust the disabled styles, we will have the issue with checking the checkbox. You cannot check a disabled checkbox

**Comment by MajaZarkova** [comment]
Other workarounds that I've tried and didn't work:

- https://stackoverflow.com/questions/74571705/prevent-child-elements-from-being-interactive-and-focusable 
- role="presentation"
- role="none"

**Comment by JoCa96** [comment]
IMHO, these are false positives according to the pattern https://www.w3.org/WAI/ARIA/apg/patterns/combobox/:
- The scroll container should not be focusable, as it is scrolled/controlled through the keyboard actions
- The nested checkboxes are also controlled by as described
